django_airavata/apps/workspace/static/django_airavata_workspace/js/containers/UserStorageContainer.vue

```diff
@@ -10,43 +10,60 @@
       @delete-file="deleteFile"
       @directory-selected="directorySelected"
       @file-content-changed="fileContentChanged"
+      :allow-preview="false"
     ></router-view>
   </div>
 </template>
 
 <script>
-import { services, utils } from "django-airavata-api";
-import { notifications } from "django-airavata-common-ui";
+import {services, utils} from "django-airavata-api";
+import {notifications} from "django-airavata-common-ui";
 
 export default {
   name: "user-storage-container",
-  computed: {
-    storagePath() {
-      let _storagePath = /~.*$/.exec(this.$route.fullPath);
-      if (_storagePath && _storagePath.length > 0) {
-        _storagePath = _storagePath[0];
+  computed: {},
+  data() {
+    return {
+      dataProductUri: null,
+      storagePath: null,
+      userStoragePath: null
+    };
+  },
+  methods: {
+    setDataProductUri() {
+      let _dataProductUri = /\?.*dataProductUri=(.*)/.exec(window.location.href);
```

Comment:
Why is this needed? This seems like a weird way to get the data product URI.

Reply:
Okay, I think I see. Keep in mind you can pass props to routes, that might be a better way to pass the dataProductUri https://router.vuejs.org/guide/essentials/passing-props.html. In fact you can keep the old behavior of the routing being based on the path but pass the dataProductUri as a prop.

Reply:
Changed it to `this.$route.query.dataProductUri`